storage/src/mmr/mem.rs

```diff
@@ -142,30 +184,46 @@ impl<H: CHasher> Mmr<H> {
         Proof::<H>::range_proof::<Mmr<H>>(self, start_element_pos, end_element_pos).await
     }
 
-    /// Returns the position of the oldest element that must be retained by this MMR in order to
-    /// preserve its ability to generate proofs for new elements. This is the position of the
-    /// tallest peak.
-    fn oldest_required_element(&self) -> u64 {
-        match self.peak_iterator().next() {
-            None => {
-                // Degenerate case, only happens when MMR is empty.
-                0
+    /// Forget all but the very last node (and any peak).
+    pub fn forget_all(&mut self) {
+        if !self.nodes.is_empty() {
+            self.forget_to_pos(self.index_to_pos(self.nodes.len() - 1));
+        }
+    }
+
+    /// Forget all nodes up to but not including the given position (except for any peaks in that
+    /// range).
+    ///
+    /// Forgotten nodes will no longer be provable, nor will some nodes that follow them in some
+    /// cases.  Use forget_max() to guarantee a desired node (and all that follow it) will remain
+    /// provable after forgetting.
+    pub fn forget_to_pos(&mut self, pos: u64) {
```

Comment:
I wonder if we want this one to be public (could be called `forget_unchecked`). Seems better to corral users to either `forget_all` (which I think should be called `forget_max`) or `forget_max` (which I think should be called `forget_to_pos`).

Reply:
I find `forget_all` to be more descriptive/literal than `forget_max` for the case where we forget everything.  I think `forget_max` may just be too ambiguous. Still trying to think of a better name though...

Reply:
Alright I've changed "forget_max" to just plain old "forget", left forget_to_pos named as is but reduced its visiblity to internal to the crate only.  WDYT?

Reply:
I think its definitely an improvement. 👍 

Not sure how I feel about `forget_all` vs `forget_max` but that's your call. Definitely "good enough" now.